code/modules/admin/sql_ban_system.dm

```diff
@@ -3,54 +3,88 @@
 
 #define MAX_REASON_LENGTH 600
 
-//checks client ban cache or DB ban table if ckey is banned from one or more roles
-//doesn't return any details, use only for if statements
-/proc/is_banned_from(player_ckey, list/roles)
-	if(!player_ckey)
-		return
+/**
+ * Checks client ban cache or, if it doesn't exist, queries the DB ban table to see if the player's
+ * ckey is banned from at least one of the provided roles.
+ *
+ * Returns TRUE if the player matches with one or more role bans.
+ * Returns FALSE if the player doesn't match with any role bans. Possible errors states also return FALSE.
+ *
+ * Args:
+ * * player_key - Either key or ckey of the player you want to check for role bans.
+ * * roles - Accepts either a single role string, or a list of role strings.
+ */
+/proc/is_banned_from(player_key, list/roles)
+	if(!player_key)
+		stack_trace("Called is_banned_from without specifying a ckey.")
```

Comment:
It seems adding an antag datum to a (c)keyless mon calls `is_banned_from`, as seen in the traitor unit test. If you wish to keep this stack_trace, remember to take care of that. Otherwise, lgtm.

Reply:
Feexed